### PR TITLE
Remove editor-specific resource rendering

### PR DESCRIPTION
**Pull request by pchote**
This PR implements an alternative approach to #18403. This solves the problem of the map editor's custom resource rendering code not supporting multiple layers by removing the map editor's custom resource rendering code completely. The game and editor resource layers now implement a common `IResourceLayer` interface that allows the ingame renderer to work for both.

This PR requires modders to move their `ResourceRenderer` definitions from `World` to `^BaseWorld`, which cannot be automated using an update rule. This will need to be mentioned in the SDK update notes.

This also fixes a crash when mods try to actually use multiple renderers ingame. Further improvements that I had planned to support TS vein rendering are deferred until a much larger rework of the way resources are defined has been completed.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Harvester.cs [hunk 1]
```diff
@@ -380,10 +380,12 @@ public bool CanTarget(Actor self, Target target, List<Actor> othersAtTarget, ref
 				if (!self.Owner.Shroud.IsExplored(location))
 					return false;
 
-				var res = self.World.WorldActor.Trait<ResourceRenderer>().GetRenderedResourceType(location);
 				var info = self.Info.TraitInfo<HarvesterInfo>();
+				var res = self.World.WorldActor.TraitsImplementing<ResourceRenderer>()
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Harvester.cs:384]: How often is `CanTarget` called (for one active targeter) and would it make sense to cache the trait lookup in the ctor?

Comment (pchote) [reply OpenRA.Mods.Common/Traits/Harvester.cs:384]: I expect that the targeter is recreated each tick so caching it there won't actually win us anything. We could cache it in `Harvester` and pass it to the OT ctor if we think / profile it to be worthwhile?

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/Harvester.cs:384]: It's probably not worth then.

**Review by abcdefg30 — APPROVED** [review]
Didn't spot regressions ingame.

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/5c4c5c5e636eb8873a6be4de1bbb3a748d5f8958...53cffc43a0b532555f2b5d24ac0ca23044e7d641

**Pull request merged**